lib/source-maps.js

```diff
@@ -4,14 +4,20 @@ const libSourceMaps = require('istanbul-lib-source-maps')
 const fs = require('fs')
 const path = require('path')
 
-const sourceMapCache = libSourceMaps.createSourceMapStore()
+let sourceMapCache = libSourceMaps.createSourceMapStore()
 function SourceMaps (opts) {
   this.cache = opts.cache
   this.cacheDirectory = opts.cacheDirectory
   this.loadedMaps = {}
   this._sourceMapCache = sourceMapCache
 }
 
+SourceMaps.prototype.purgeCache = function () {
+  sourceMapCache = libSourceMaps.createSourceMapStore()
```

Comment:
I don't understand why we'd need to throw out source maps like this,  might be worth digging into things a bit more, and figuring out how we end up with incompatible source-maps in the first place.

I'm not opposed to landing this as a patch  as a temporary measure, but it feels like we're continuing to complicate our source-map and caching logic, which is already a mess. I think you're right that we just need to get way more test cases in the codebase.

Reply:
Same reason we need to throw out transform results (they set `cache: false`).